### PR TITLE
Make it possible to generate GAP functions from Julia

### PR DESCRIPTION
**Pull request by sebasguts**

### CODE REVIEW

**Comment by sebasguts** [comment]
This allows for:
```
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.1.0 (2019-01-21)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using GAP
Adding path /home/sebastian/Software/GAPJulia/gap/.libs to DL_LOAD_PATH
 ┌───────┐   GAP 4.dev of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-julia64-kv6
 Configuration:  gmp 6.1.2, Julia GC, Julia 1.1.0, readline
 Loading the library and packages ...
^[[B Packages:   GAPDoc 1.6.1, PrimGrp 3.3.1, SmallGrp 1.3, TransGrp 2.0.4
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'

julia> oper = (i,g)->i^g
#3 (generic function with 1 method)

julia> GAP.Globals.Orbit(GAP.Globals.SymmetricGroup(3),1,GAP.julia_to_gap(oper))
GAP: [ 1, 2, 3 ]
```

**Comment by fingolfin** [comment]
Please rebase so that CI runs again and we get proper coverage.

**Comment by sebasguts** [comment]
Done.

**Comment by fingolfin** [comment]
Clearly something is still broken wrt coverage, the C code has no coverzreported (and hence this PR shows no coverage).

**Comment by fingolfin** [comment]
Am on a phone only now, but: do we still oasd --coverage in CFLAGS and LDFLAGS when compiling our code? (Possibly indirectly, if GAP was compiled with those flags)

**Comment by sebasguts** [comment]
@fingolfin We don't for the GAP compiled by julia, I assume. Should I add this simply to the debug mode? Guess it won't hurt there.

File: test/conversion.jl [hunk 1]
```diff
@@ -226,4 +226,12 @@ end
     xx = GAP.EvalString( "[1,,1]" )
     @test GAP.julia_to_gap([1,nothing,1]) == xx
 
+    ## Test function conversion
+    return_first(args...) = args[1]
+    return_first_gap = GAP.julia_to_gap(return_first)
+    @test GAP.Globals.IsFunction(return_first) == false
+    @test GAP.Globals.IsFunction(return_first_gap) == true
+    list = GAP.EvalString("[1,2,3]")
+    @test GAP.Globals.List(list,return_first_gap) == list
```
Comment (fingolfin) [inline test/conversion.jl]:
```suggestion
    @test GAP.Globals.List(list, return_first_gap) == list
```

**Pull request merged**